functionsandLoops/funcandloops_ex.py

```diff
@@ -1,8 +1,79 @@
+# ex1 - the function prints all the numbers from 1 to n in seperate lines and print the function's return value
 def func(n: int):
     for i in range(n):
         print (i)
     return "finished"
 
+# ex2 - function which prints 3 different sequences
+def fun(n: int):
+    # first required sequence
+    for i in range(n+1):
+        # converting the first element into '1' 
+        if i == 0:
+            print("First sequence:\n", 1, end = " ")
+        # the other elements are the even ones 
+        if i%2 == 0 and i != 0:
+            print(i, end = " ")
+    print ("\n", end = " ")
+    # second required sequence 
+    for i in range(n+1):
+        # converting the first element into '1' 
+        if i == 0:
+            print("Second sequence:\n", 1, end = " ")
+        # the other elements are the powerset of 2
+        elif 2**i <= (n+1):
+            print(2**i, end = " ")
+    print ("\n", end = " ")
+    # first two elements in the required sequence
+    n1, n2 = 1, 1
+    # check if the number typed is valid
+    if n <= 0:
+        print("Please enter a positive integer")
+    # if the int equals to 1, return 1 only
+    elif n == 1:
+        print("Fibbonacci sequence upto", n, ":")
+        print(n1)
+    # generate fibonacci sequence
+    else:
+        print("Third sequence:\n", end = " ")
+        while n1 < (n+1):
+            print(n1, end = " ")
+            nsm = n1 + n2
```

Comment:
Improve your names! in this case 'temp' would be good instead of nsm.
Temp represents a variable which is used temporarily, as made in this case.